tests/drivers/gpio/egpio_more_loops/src/main.c

```diff
@@ -0,0 +1,746 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "main.h"
+
+/* Assumption: All input GPIOs are on the same device */
+/* Assumption: All output GPIOs are on the same device */
+static const struct device *const dev_out = DEVICE_DT_GET(DEV_OUT);
+static const struct device *const dev_in = DEVICE_DT_GET(DEV_IN);
+
+/* Delay after setting GPIO ouputs. It allows signals to settle.
+ */
+#define PULL_DELAY_MS K_MSEC(1U)
+
+/* Check that input GPIOs state match with input parameter 'value'. */
+static void _check_inputs(uint32_t value)
+{
+	int rc;
+	gpio_port_value_t v1;
+	bool current, expected;
+
+	/* stabilize logic level */
+	k_sleep(PULL_DELAY_MS);
+
+	/* get input port state */
+	rc = gpio_port_get_raw(dev_in, &v1);
+	zassert_equal(rc, 0, "gpio_port_get_raw(dev_in) failed");
+
+	/* Check each bit has expected state
+	 * MSB [IN_5, IN_4, IN_3, IN_2, IN_1] LSB
+	 */
+	current = (v1 & BIT(PIN_IN_1)) ? true : false;
+	expected = (value & BIT(0)) ? true : false;
+	zassert_equal(current, expected, "IN_1 = %d, while expected %d", current, expected);
+	current = (v1 & BIT(PIN_IN_2)) ? true : false;
+	expected = (value & BIT(1)) ? true : false;
+	zassert_equal(current, expected, "IN_2 = %d, while expected %d", current, expected);
+	current = (v1 & BIT(PIN_IN_3)) ? true : false;
+	expected = (value & BIT(2)) ? true : false;
+	zassert_equal(current, expected, "IN_3 = %d, while expected %d", current, expected);
+	current = (v1 & BIT(PIN_IN_4)) ? true : false;
+	expected = (value & BIT(3)) ? true : false;
+	zassert_equal(current, expected, "IN_4 = %d, while expected %d", current, expected);
+	current = (v1 & BIT(PIN_IN_5)) ? true : false;
+	expected = (value & BIT(4)) ? true : false;
+	zassert_equal(current, expected, "IN_5 = %d, while expected %d", current, expected);
+}
+
+/* Set output GPIOs with gpio_pin_configure() and confirm that input GPIOs have expected state. */
+static void _gpio_pin_configure_and_check(uint32_t value)
+{
+	int rc;
+
+	/* OUT_1 */
+	/* Disconnect output */
+	rc = gpio_pin_configure(dev_out, PIN_OUT_1, GPIO_DISCONNECTED);
+	if (rc == -ENOTSUP) {
+		TC_PRINT(
+			"gpio_pin_configure(PIN_OUT_1, GPIO_DISCONNECTED) failed; trying with GPIO_INPUT\n");
```

Comment:
What is being tested here?

Reply:
I copied this from here:
https://github.com/nrfconnect/sdk-nrf/blob/main/tests/drivers/gpio/egpio_basic_api/src/test_egpio_port.c#L105

My understanding is that:
- before configuring pin as GPIO_OUTPUT_HIGH / GPIO_OUTPUT_LOW try to configure pin as disconnected;
- if implementation doesn't support GPIO_DISCONNECTED, try with GPIO_INPUT.
- if both GPIO_DISCONNECTED and GPIO_INPUT have failed, report error.